### PR TITLE
Refactor and add getIndexUnsafe

### PR DESCRIPTION
**Pull request by harendra-kumar**
Review commit by commit, first commit is rearrangement of code, second is enhancement.

### CODE REVIEW

File: src/Streamly/Internal/Data/Array.hs [hunk 1]
```diff
@@ -193,14 +229,14 @@ fold f arr = D.fold f (toStreamD arr)
...
+-- Random reads and writes
+-------------------------------------------------------------------------------
+
+-- | /O(1)/ Lookup the element at the given index. Index starts from 0. Does
+-- not check the bounds.
+--
+-- @since 0.8.0
+{-# INLINE getIndexUnsafe #-}
+getIndexUnsafe :: Array a -> Int -> a
+getIndexUnsafe = indexArray
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/Array.hs:242]: `indexArray` comes from `primitive`?

**Review by adithyaov — APPROVED** [review]
LGTM, Please fix the CI before the merge.

**Comment by harendra-kumar** [comment]
None of the CIs have a fixable error.

File: src/Streamly/Internal/Data/Array.hs (hunk 1 above)
Comment (harendra-kumar) [reply src/Streamly/Internal/Data/Array.hs:242]: Yes.

**Pull request merged**